src/Data/Foreign/Class.purs

```diff
@@ -64,6 +69,17 @@ instance undefinedIsForeign :: (IsForeign a) => IsForeign (Undefined a) where
 instance nullOrUndefinedIsForeign :: (IsForeign a) => IsForeign (NullOrUndefined a) where
   read = readNullOrUndefined read
 
+instance maybeIsForeign :: (IsForeign a) => IsForeign (Maybe a) where
+  read f
+    | isNull f  = pure Nothing
+    | otherwise = (Just <$> read f) <|> (pure Nothing)
+
+instance isForeignTuple :: (IsForeign a, IsForeign b) => IsForeign (Tuple a b) where
```

Comment:
I think this instance might be better provided as a function, rather than an instance, since an instance canonicalizes the behavior for the type, and this behavior is quite special.